File: zad1/node/zad1.js
const input = process.argv[2].split('');

const solveCaptcha = (input, distance) => {
  let sum = 0;
  for (let i = 0; i < input.length; i++) {
    const first = input[i];
    const second = input[(i + distance) % input.length];
    if (first === second) {
      sum += +first;
    }
  }
  return sum;
};

console.log("Part 1 : " + solveCaptcha(input, 1));
console.log("Part 2 : " + solveCaptcha(input, input.length / 2));

File: zad1/python/zad1.py
import sys

def solve_captcha(input, distance):
    sum = 0
    for i in range(0, len(input)):
        first = input[i]
        second = input[(i + distance) % len(input)]
        if first == second :
            sum += int(first)
    return sum

arg = str(sys.argv[1])

print("Part 1 : " + str(solve_captcha(arg, 1)))
print("Part 2 : " + str(solve_captcha(arg, int(len(arg) / 2))))